server/consumer.go

```diff
@@ -3731,7 +3731,8 @@ func (o *consumer) processInboundAcks(qch chan struct{}) {
 	// How often we will check for ack floor drift.
 	// Spread these out for large numbers on a server restart.
 	delta := time.Duration(rand.Int63n(int64(time.Minute)))
-	var ackFloorCheck = time.Minute + delta
+	ticker := time.NewTicker(time.Minute + delta)
```

Comment:
I think that to replace this timer we have to do it inside of the `for` before `select` like:

```go
	for {
		ticker := time.NewTimer(time.Minute + delta)
		select {
		case <-o.ackMsgs.ch:
			ticker.Stop()
			// ...
		case <-ticker.C:
			o.checkAckFloor()
		case <-qch:
			t.Stop()
			return
		case <-s.quitCh:
			t.Stop()
			return
		}
```

Reply:
I made it into a Ticker vs one shot timers. Always should be firing and checking so made single Ticker outside for loop.